codalab/lib/worksheet_util.py

```diff
@@ -460,6 +460,8 @@ def interpret_file_genpath(client, target_cache, bundle_uuid, genpath, post):
                     except:
                         # Plain text file
                         info = ''.join(contents)
+            if not info: # info is an empty dictionary
+                info = None
```

Comment:
In general, we want `info` to preserve information about the file, so don't want to just set it to None.  What's happening here is that `contents = []`.  In this case, we shouldn't be interpreting this as a TSV file, but a plain text file, which means, info should be set to the empty string.  I would add this as the first condition before line 446.  Python makes it very easy to be sloppy with types; it's good to always think about the semantics - difference between `{}, [], '', None`.